packages/seal/src/Search/Condition/AndCondition.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Schranz\Search\SEAL\Search\Condition;
+
+class AndCondition
+{
+    /**
+     * @var array<object>
+     */
+    public function __construct(public readonly array $conditions)
```

Comment:
What you think is more developer friendly go here with an `array` or variable parameters length e.g.:

```php
public function __construct(...$conditions)
```

```php
$condition = new Condition\OrCondition(
    new Condition\AndCondition(
        new Condition\EqualCondition('tags', 'Tech'),
        new Condition\EqualCondition('isSpecial', true),
    ),
    new Condition\AndCondition(
        new Condition\EqualCondition('tags', 'UX'),
        new Condition\EqualCondition('isSpecial', false),
    ),
);
```